Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -3874,4 +3898,62 @@ public void addExtraInstrumentationData(HashMap<String, String> instrumentationD
     public void addExtraInstrumentationData(String key, String value) {
         instrumentationExtraData_.put(key, value);
     }
+
+
+    //-------------------- Promo view handling--------------------//
+
+
+    @Override
+    public void onPromoViewVisible(String action) {
+        //No Implementation on purpose
+    }
+
+    @Override
+    public void onPromoViewDismissed(String action) {
+        if(ServerRequestInitSession.isInitSessionAction(action)){
+            checkForAutoDeepLinkConfiguration();
+        }
+    }
+
+    @Override
+    public void onPromoViewError(int errorCode, String errorMsg) {
+
+    }
+
+    /**
+     * Interface for defining optional promo view behaviour for Activities
+     */
+    public interface IPromoViewControl{
+        /**
+         * Defines if an activity is interested to show promo views or not.
+         * By default activities are considered as promo view enabled. In case of activities which are not interested to show a promo view (Splash screen for example)
+         * should implement this and return false. The pending promo view will be shown with the very next promo view activity
+         *
+         * @return A {@link Boolean} whose value is true if the activity don't want to show any promo view.
+         */
+        boolean skipPromoViewsOnThisActivity();
+    }
+
+    //--------- Test data to simulate Promo view------------//
+    //TODO should be part of initSession JsonObj. Delete as back end implement this feature.
+    public JSONArray getPromoViewData() {
+        JSONArray promoViewArray  = null;
+        String previewArrayObj = "{\"app_promo_data\":[{"
+                + " \"app_promo_id\":\"promo_id_01\","
+                + " \"app_promo_action\" : \"open\","
+                + " \"num_of_use\":1,"
+                + " \"promo_view_url\":\"https://branch.io\","
+                + " \"expiry\":123456778 },{"
+                + " \"app_promo_id\":\"promo_id_02\","
+                + " \"app_promo_action\" : \"buy\","
+                + "  \"num_of_use\":1,"
+                + " \"promo_view_url\":\"https://branch.io\","
```

Comment:
Since we're not doing the backend piece yet, I'm a bit concerned we'll find that prepping/sending html blobs in the API is too laggy. This might be me being too paranoid. However, let's just support both to be safe.